Description: work around a GCC bug that results in build failures on i386 and s390x
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Bug: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90756
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=930012

--- a/gfx/skia/skia/third_party/skcms/src/Transform_inl.h
+++ b/gfx/skia/skia/third_party/skcms/src/Transform_inl.h
@@ -559,7 +559,7 @@ SI void sample_clut_16(const skcms_A2B*
 
 // GCC 7.2.0 hits an internal compiler error with -finline-functions (or -O3)
 // when targeting MIPS 64,  I think attempting to inline clut() into exec_ops().
-#if 1 && defined(__GNUC__) && !defined(__clang__) && defined(__mips64)
+#if 1 && defined(__GNUC__) && !defined(__clang__) && (defined(__mips64) || defined(__i386) || defined(__s390x__))
     #define MAYBE_NOINLINE __attribute__((noinline))
 #else
     #define MAYBE_NOINLINE
